src/app/auth/profiles/profiles.component.ts

```diff
@@ -91,12 +92,15 @@ export class ProfilesComponent implements OnInit {
   }
 
   /**
-   * Processes available Profiles information from application's configuration.
+   * Processes available Profiles information from the extenral repository.
    */
   initProfiles(): void {
-    this.profiles = this.profiles
-      .concat(AppConfig.profiles)
+    this.profileService.fetchExternalProfiles().then(externalProfiles => {
+      this.profiles = this.profiles
+      .concat(externalProfiles)
       .filter((p) => !!p);
+    })
+    .catch(e => this.openErrorDialog());
```

Comment:
The error dialog only applies for malformed profiles' file, but we're launching it for all errors.
I think, for other errors (e.g. network issue), we should pass the error to ErrorHandlingService.

Reply:
:+1:

Reply:
Will refactor it to the logic flow as below! 

```typescript 
    .catch(e => {
      if (e === MALFORMED_PROFILES_ERROR) {
        this.openErrorDialog();
      } else {
        this.errorHandlingService.handleError(e);
      }
    });
```